src/mito2/src/manifest/storage.rs

```diff
@@ -220,10 +192,32 @@ impl ManifestObjectStore {
 
         entries.sort_unstable_by(|(v1, _), (v2, _)| v1.cmp(v2));
 
-        Ok(ObjectStoreLogIterator {
-            object_store: self.object_store.clone(),
-            iter: Box::new(entries.into_iter()),
-        })
+        Ok(entries)
+    }
+
+    /// Fetch all manifests in concurrent.
+    pub async fn fetch_manifests(
+        &self,
+        manifests: &[(u64, Entry)],
+    ) -> Result<Vec<(ManifestVersion, Vec<u8>)>> {
+        let tasks = manifests.iter().map(|(v, entry)| async {
+            let compress_type = file_compress_type(entry.name());
+            let bytes = self
+                .object_store
+                .read(entry.path())
+                .await
+                .context(OpenDalSnafu)?;
+            let data = compress_type
+                .decode(bytes)
+                .await
+                .context(DecompressObjectSnafu {
+                    compress_type,
+                    path: entry.path(),
+                })?;
+            Ok((*v, data))
+        });
+
+        try_join_all(tasks).await
```

Comment:
do we need to limit the maximum concurrency?

Reply:
@evenyag Any suggestions? How about setting the limit to 8?

Reply:
how about using `manifest_checkpoint_distance`(default is 10) to make dynamic decision, `manifest_checkpoint_distance` decides the max number of manifest files.
 if the limit is 16, the limit doesn't work in this case because the max number is 10.

Reply:
Cool, It seems we don’t need this limit here?

Reply:
Or it should be `min(manifest_checkpoint_distance,limit)`

Reply:
I decided to remove the limit. It typically wouldn't have too many concurrent requests 🥹.